### DISCUSSION TITLE
--netlock: Error: no valid sandbox

### DISCUSSION THREAD

**Discussion by Dieterbe** [netblue30/firejail#5395] [Q&A] [answered]
Hello!
I'm interested in using https://logseq.com/, which is "personal knowledgement software" (a fancy note taking app).  Now, before I start putting in some personal notes, it seemed wise to lock down its networking abilities, such that I know that my personal notes will remain personal.
However, the app is able to embed tweets and youtube videos within notes, so obviously I want that to work.
As I was reading `man firejail` and stumbled upon the `netlock` option, a brilliant idea came to mind.  Possibly the most brilliant idea of my life.

I could just wrap the logseq program in a shellscript, do some approved sample network traffic (downloading some data from twitter and youtube), sleep 60 seconds, and then start the program.  Netlock will then set up the firewall rules exactly how I like, right? A perfectly elegant solution, and not hacky at all. At least in my mind.  Happy to trade 60 seconds of productivity for increased security. 

[Here](https://gist.github.com/Dieterbe/216185e2d87b5b20548e56de3a943150) is the profile, script and output.
The main clue is probably this:
```
Error getpwuid: main.c:243 init_cfg: No such file or directory
Error: no valid sandbox
```

despite the errors, the app runs and shows up in `firejail --list`
```
44811:dieter::firejail --profile=/home/dieter/.config/firejail/logseq.profile --netlock sh /home/dieter/logseq-wrapper
```
But all network traffic seems completely blocked. trying to embed that tweet in a note shows nothing.

Is there a way to debug the config that netlock builds and applies? What's the error about? Is my idea really brilliant or ludicrous?

**Comment by rusty-snake** [comment]
Start `wget` by full path. It has a firecfg symlink too.

**Comment by rusty-snake** [comment]
> Is my idea really brilliant or ludicrous?

If it works for you, nothing wrong. 

Alternative feature: 
- fdns
- `netfilter`

Keep in mind that twitter/youtube are served via CDNs that host several domain on the sam IP addresses.

**Comment by Dieterbe** [comment]
> Start wget by full path. It has a firecfg symlink too.

Good call!


So I'm noticing:
1) the "Error: no valid sandbox" seems unrelated to my wrapper. I see it also with the command:
```
firejail --profile=/home/dieter/.config/firejail/logseq.profile --netlock /usr/bin/logseq
```
However, the error disappears by simply removing the `--netlock` flag.
I have attached the logs of the run with and without the flag. The only difference between them is the error appears when netlock is active.

[netlock.txt](https://github.com/netblue30/firejail/files/9697663/netlock.txt)
[without-netlock.txt](https://github.com/netblue30/firejail/files/9697665/without-netlock.txt)

2)`wget: unable to resolve host address ‘twitter.com’`, this happens consistently. my config is below. enabling/disabling netfilter has no effect. I suspect this is due to 1)
```
$ grep net .config/firejail/logseq.profile 
#netfilter
#net none
net enp0s13f0u4u3u1
#netfilter
```

**Reply by rusty-snake** [reply]
1. Search for the error message in the issue tracker: #5312

**Reply by rusty-snake** [reply]
2. If you hide resolv.conf, how should wget know the DNS?
   Replace `private-etc none` with `private-etc resolv.conf`. However this is not enough if you want to use TLS protected connection (as tw/yt will enforce I guess). https://github.com/netblue30/firejail/blob/393c5beff2686d7732221dadb6730917f24835a0/etc/templates/profile.template#L187-L198

**Comment by netblue30** [comment, marked as answer]
@Dieterbe - we had a bug in the last version, something very similar. It is fixed on mainline: https://github.com/netblue30/firejail/issues/5312

**Comment by YuAnKj0** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。
